WalletWasabi.Fluent/ViewModels/Wallets/CoinJoinStateViewModel.cs

```diff
@@ -128,56 +142,38 @@ public CoinJoinStateViewModel(WalletViewModel walletVm, IObservable<Unit> balanc
 				_stateMachine.Fire(Trigger.PlebStopChanged);
 			});
 
+		_autoCoinJoinStartTimer = new DispatcherTimer { Interval = TimeSpan.FromMinutes(Random.Shared.Next(5, 16)) };
+		_autoCoinJoinStartTimer.Tick += async (_, _) =>
+		{
+			await coinJoinManager.StartAsync(_wallet, stopWhenAllMixed: false, false, CancellationToken.None);
+			_autoCoinJoinStartTimer.Stop();
+		};
```

Comment:
Im sure this works but it smells hacky. Why not do it as usual?

```c#
Task AutoCoinJoinStartTask() => new (async () =>
{
     await Task.Delay(randomTime);
     await coinJoinManager.StartAsync(_wallet, stopWhenAllMixed: false, false, CancellationToken.None);
});
```

and then you should be able to start it with:

```c#
AutoCoinJoinStartTask().Start();
``` 

it's just an idea.

Reply:
Could be. I noticed that disposing of things here is problematic -  I did a similar pattern like above.